docs/architecture.rst

```diff
@@ -1,142 +1,211 @@
+    construct the function or class in a particular worker and assign it a name in the key-value store, perhaps
+    overwriting an existing object.
+4. **Calling the Function or Class**: After deploying the function, class, or object into the server, the Runhouse
+    Python client returns a local callable stub which behaves like the original object, but forwards method calls
+    over HTTP to the remote object on the cluster.
+        a. If a stateful instance of a class is desired, an `__init__` method can be called on the remote class to
+            instantiate a new remote object from the class and assign it a name.
+        b. If arguments are passed to the method, they're serialized with cloudpickle and sent with the HTTP request.
+            Serializing code, such as functions, classes, or dataclasses, is strongly discouraged, as it can lead to
+            versioning mismatch errors between local and remote package versions.
+        c. From here on, you can think of Runhouse as facilitating
+            regular object-oriented programming but with the objects living remotely, maybe in a different cluster,
+            region, or cloud than the local code.
+        d. Python behavior like async, exceptions, printing, and logging are all preserved across remote calls, but
+            can be disabled or controlled if desired.
+5. **Saving and Loading**: The Runhouse client can save and load objects to and from the local filesystem, or to a
+    remote metadata store. This allows for easy sharing of clusters and services across users and environments,
+    and for versioning and rollback of objects. The metadata store can be accessed from any Python interpreter,
+    and is backed by a management API to view and manage all resources.
+6. **Terminating Modules, Workers, or Clusters**: When a remote object is no longer needed, it can be deallocated from
+    the remote compute by calling `cluster.delete(obj_name)`. This will remove the object from the key-value store and
+    free up the memory on the worker. A full worker can similarly be terminated with `cluster.delete(worker_name)`. A
+    cluster can be terminated with `cluster.down()`.
+
+Comparing to other systems
+~~~~~~~~~~~~~~~~~~
 
-* `Runhouse Den <https://www.run.house/dashboard>`_: provides an individual or admin view of all resources, secrets,
-  groups, and sharing. Resource metadata is automatically versioned in RNS, allowing teams to maintain single-sources
-  of truth for assets with zero downtime to update or roll back, and trace exact lineage for any existing resource.
+Runhouse's APIs bear similarity to other systems, so it's helpful to compare and contrast. In many cases,
+Runhouse is not a replacement for these systems, but rather a complement or extension. In others, you may be able
+to replace your usage of the other system entirely with Runhouse.
+
+*Distributed frameworks* like Ray, Spark, or Elixr make it possible to offload execution onto separate
```

Comment:
These could also potentially be h3s to break up the section and improve readability